### ISSUE TITLE
KerasR time serie LSTM multidimensional

### ISSUE THREAD

**Issue by dma100180** [rstudio/keras#270]
Hello, I following one of the example (https://github.com/rstudio/keras/blob/master/vignettes/examples/stateful_lstm.R), I changed the data and it works for me, but when I only received one input measure, when I try to do it with two I get an error:
Error when checking input: expected lstm_8_input to have 3 dimensions, but got array with shape (1203, 1203, 1, 1)

[data.zip](https://github.com/rstudio/keras/files/1691910/data.zip)

DataKeras<-read.csv(file="D:/data.csv")
DataKeras
dak<-DataKeras[,c(3,4,50)]

trx <- array(data = c(dak[,1],dak[,2]), dim = c(length(dak[,1]),length(dak[,1]), 1, 1))
try <- array(data = dak[,3], dim = c(length(dak[,3]), 1))

library(keras)
tsteps <- 1
batch_size <- 3
epochs <- 25
lahead <- 2

model <- keras_model_sequential()
model %>%
  layer_lstm(units = 50, input_shape = c(2,1), batch_size = batch_size,
             return_sequences = TRUE, stateful = TRUE) %>% 
  layer_lstm(units = 50, return_sequences = FALSE, stateful = TRUE) %>% 
  layer_dense(units = 1)
model %>% compile(loss = 'mse', optimizer = 'rmsprop')

for (i in 1:epochs) {
  model %>% fit(trx, try, batch_size = batch_size,
                epochs = 1, verbose = 1, shuffle = FALSE)
  
  model %>% reset_states()
}
predicted_output <- model %>% predict(trx, batch_size = batch_size)

Thank you!!

**Comment by dma100180** [comment]
DataKeras<-read.csv(file="D:/data.csv")

trx <- array(data = c(dak[,1],dak[,2]), dim = c(length(dak[,1]), 2, 1))
try <- array(data = dak[,3], dim = c(length(dak[,1]), 1))

library(keras)
tsteps <- 1
batch_size <- 3
epochs <- 25
lahead <- 1

model <- keras_model_sequential()
model %>%
  layer_lstm(units = 50, input_shape = c(2,1), batch_size = batch_size,
             return_sequences = TRUE, stateful = TRUE) %>% 
  layer_lstm(units = 50, return_sequences = FALSE, stateful = TRUE) %>% 
  layer_dense(units = 1)
model %>% compile(loss = 'mse', optimizer = 'rmsprop')


for (i in 1:epochs) {
  model %>% fit(trx, try, batch_size = batch_size,
                epochs = 1, verbose = 1, shuffle = FALSE)
  
  model %>% reset_states()
}

predicted_output <- model %>% predict(trx, batch_size = batch_size)

**Comment by dma100180** [comment]
Hi again, the problem was in the definition of the input array trx, which delimits the length, the number of input variables for the prediction and the number of variables to predict in the third component, in my case it is [1023, 2.1] since I have increased in 1 the predictive predictors from 1 to 2 leaving the variables to predict in 1.
In the model, in input_shape = (2,1) which corresponds to the predicted dimensions and the dimensions to be predicted. The second message is the correct code

I hope this helps more people. a greeting

**Comment by Steviey** [comment]
very useful tip, thank you!
